components/text/UIDateInput/index.js

```diff
@@ -112,15 +111,21 @@ export default class UIDateInput extends UIDetailsInput<Props, State> {
 
     isValidDate() {
         const date = this.getValue();
-        const validDate = Moment(date, this.getPattern()).isValid();
+        const validDate = Moment(date, this.getPattern(true)).isValid();
         const validLength = date.length === UIConstant.shortDateLength() || date.length === 0;
 
         return (validDate && validLength);
     }
 
-    getPattern() {
+    // This method returns a localized string that will be rendered
+    // in order to indicate the date component that has to be entered.
+    // However, internally (programming language), in order to parse/localized
+    // a date, patter symbols have to be provided in English.
+    getPattern(internalPattern: boolean = false) {
```

Comment:
Maybe to change the naming? since it's not very clear what is "internal"...
I would suggest to add another opposite parameter called `localizedPattern` and use it in those methods which don't require **Moment**, it'd be much easier to read as I think...